enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+
+- [Introduces the upstream Bundle and BundleDeployment OLM 1.x v1alpha1 API(s)](https://github.com/operator-framework/rukpak/tree/main/api/v1alpha1).
+- [Introduces a validating webhook for the Bundle API](https://github.com/operator-framework/rukpak/blob/main/api/v1alpha1/bundle_webhook.go).
+
+### Risks and Mitigations
+
+#### Risks
+
+- Early phases will introduce multiple v1alpha1 CRDs downstream.
+- Early phases will introduce OLM v1 components that are in alpha stages.
+- No concrete migration plan from legacy to OLM v1.
+- Coordination around external team dependencies.
+- The BundleDeployment API has a large net of permissions.
+- The upstream rukpak project has a direct dependency on cert-manager.
+
+#### Mitigations
+
+The v1alpha1 CRDs won't be deployed by default, and those APIs will be hidden behind the "TechPreviewNoUpgrade" feature set that admins need to explicitly enable before cluster creation. Any new components introduced during these early phases will also specify the `release.openshift.io/feature-gate` annotation in their CVO manifests, and set that key's value to "TechPreviewNoUpgrade".
+This annotation will ensure that the CVO won't deploy any tech preview components.
+
+Instructing admins to configure upfront RBAC to limit the BundleDeployment API to the minimum set of users may help mitigate the plain provisioner's wide net of permissions. The process for constructing this upfront RBAC may require explicit documentation.
+
+We can mitigate the upstream rukpak project's direct dependency on the cert-manager operator by removing that dependency in favor of the downstream service-ca-operator component.
+
+### Drawbacks
+
+- Implementing this PO mechanism using legacy OLM would provide more immediate business value. That implementation is outlined below in the alternatives section.
+- Early phases will be delivered through tech preview guidelines. This leads to little customer feedback, given it precludes production clusters.
+- Early phases may introduce a new OLM 1.x control plane that has little-to-no interaction with the existing, legacy OLM 0.x control plane.
+- Early phases may introduce validating webhooks. These webhooks can impact the cluster's overall health when they're not available.
+- Early phases may introduce validating webhooks. By default, these webhooks will be deployed using an HA configuration, but may not have logic to accommodate SNO cluster environments.
+- Early phases may introduce a dynamic watching implementation for managing the underlying OLM operator resources. This has potential performance implications that need to be measured through benchmarking and/or addressed before any GA phase.
+- Future phase designs may require a large-scale rewrite of the OLM component.
+
+## Design Details
+
+### Open Questions [optional]
+
+- **Can future phase designs evaluate whether platform operators can be extended to cover things like the MAPI and CCM operators?**
```

Comment:
What exactly does this mean? Are you trying to say that we need to support a class of operator package that fulfills some role? Perhaps we want to expand the background here

Reply:
There's been a bunch of threads in this proposal around whether future PO designs can accommodate some of the install flexibility efforts. This was added as a soft tracking vehicle in the short term, and something we can re-evaluate during phase 0.

@elmiko might be able to provide some of those use cases, but I can try and take a stab at adding some background if we think it's necessary.

Reply:
At the moment, I read this and go "well why can't it do that" so a little colour may help others reading this to understand :)

Reply:
at the moment, i think our big questions are around how we could control installing POs during the bootstrap phase of cluster installation, and to that point how we could ensure that something like a Cloud Controller Manager, or Machine API actuator, could be installed at the proper times during the deployment process.

there might be additional followup questions and issues that arise from this discussion, but currently my mind goes to how could provide the infrastructure specific POs during an installations type scenario in such a manner that they could be utilized during the process.